### PR TITLE
ability to obtain releases of artist by last six month

### PR DESCRIPTION
**Pull request by monoskribt**

### CODE REVIEW

File: src/main/java/com/spotifyapi/service/impl/SpotifyServiceImpl.java [hunk 1]
```diff
@@ -32,4 +40,41 @@ public List<SpotifyArtist> getFollowedArtist() {
...
+    public List<SpotifyReleases> getReleases() {
+        List<SpotifyArtist> artists = getFollowedArtist();
+
+        List<SpotifyReleases> listOfAlbums = new ArrayList<>();
+
+        LocalDate sixMothAgo = LocalDate.now().minusMonths(6);
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+
+        for(SpotifyArtist artist : artists) {
+            Paging<AlbumSimplified> album = spotifyApi
```
Comment (syevtushok) [inline src/main/java/com/spotifyapi/service/impl/SpotifyServiceImpl.java]: How many albums you select?

File: src/main/java/com/spotifyapi/service/impl/SpotifyServiceImpl.java [hunk 2]
```diff
@@ -32,4 +40,41 @@ public List<SpotifyArtist> getFollowedArtist() {
...
+                    })
+                    .forEach(el -> {
+                        SpotifyReleases releases = new SpotifyReleases();
+                        releases.setId(el.getId());
+                        releases.setNameOfGroup(Arrays.stream(el.getArtists())
+                                .map(ArtistSimplified::getName)
+                                .collect(Collectors.joining(", ")));
+                        releases.setTitle(el.getName());
+                        releases.setDate(el.getReleaseDate());
+                        listOfAlbums.add(releases);
```
Comment (syevtushok) [inline src/main/java/com/spotifyapi/service/impl/SpotifyServiceImpl.java]: it would be nice to add a sorting

Comment (monoskribt) [reply src/main/java/com/spotifyapi/service/impl/SpotifyServiceImpl.java]: Done!

File: src/main/java/com/spotifyapi/service/impl/SpotifyServiceImpl.java (hunk 1 above)
Comment (monoskribt) [reply src/main/java/com/spotifyapi/service/impl/SpotifyServiceImpl.java]: added a limit of 7

File: src/main/java/com/spotifyapi/controller/SpotifyController.java [hunk 3]
```diff
@@ -21,4 +22,34 @@ public List<SpotifyArtist> getMyArtist() {
...
+        return spotifyService.getOfUserPlaylists();
+    }
+
+    @PostMapping("/save-releases")
+    public ResponseEntity<String> saveReleasesToPlaylist(@RequestParam ("playlistId") String playlistId) {
+        try {
+            spotifyService.saveReleasesToPlaylistById(playlistId);
+            return ResponseEntity.ok("Successfully added");
+        } catch (Exception e) {
+            return ResponseEntity.status(500).body("Something is wrong: " + e.getMessage());
```
Comment (syevtushok) [inline src/main/java/com/spotifyapi/controller/SpotifyController.java]: Don't return 500

**Review by syevtushok — APPROVED** [review]

File: src/main/java/com/spotifyapi/controller/SpotifyController.java (hunk 3 above)
Comment (monoskribt) [reply src/main/java/com/spotifyapi/controller/SpotifyController.java]: Changed that line

**Pull request merged**